lib/storekit/receipt.rb

```diff
@@ -1,19 +1,42 @@
+require 'ostruct'
+
 module StoreKit
-  class Receipt
-    attr_reader :receipt_data
+  class Receipt < OpenStruct
+    TRANSFORM = {
+       'unique_identifier' => true,
+       'product_id' => true,
+       'bid' => true,
+       'unique_vendor_identifier' => true,
+       'bvrs' => true,
+       'transaction_id' => proc { |v| v.to_i },
+       'original_transaction_id' => proc { |v| v.to_i },
+       'quantity' => proc { |v| v.to_i },
+       'item_id' => proc { |v| v.to_i },
+       'web_order_line_item_id' => proc { |v| v.to_i },
+       'original_purchase_date_ms' => proc { |v| Time.at(v.to_f / 1_000) },
+       'purchase_date_ms' => proc { |v| Time.at(v.to_f / 1_000) },
+       'expires_date' => proc { |v| Time.at(v.to_f / 1_000) },
+    }
 
-    def initialize(receipt_data)
-      @receipt_data = receipt_data
-    end
+    def self.parse receipt_data
```

Comment:
> I think that preserving the original receipt data as it comes from Apple is very important.

That is just non-ruby or non-rails approach, many gems I saw do the conversion to wrap the plain data with real ruby types, that is good.